sdk/examples/intkey_python/sawtooth_intkey/client_cli/workload.py

```diff
@@ -0,0 +1,194 @@
+
+                with self._lock:
+                    self._pending_batches[batch.header_signature] = \
+                        IntKeyState(
+                        name=key.name,
+                        stream=key.stream,
+                        value=key.value + 1)
+                self.delegate.on_new_batch(batch_id, key.stream)
+
+        else:
+            LOGGER.debug('Key %s completed', key.name)
+            self._create_new_key()
+
+    def on_batch_not_yet_committed(self):
+        self._create_new_key()
+
+    def _create_new_key(self):
+        with self._lock:
+            stream = random.choice(self._streams) if \
+                len(self._streams) > 0 else None
+
+        batch_id = None
+        if stream is not None:
+            name = datetime.now().isoformat()
+            txn = create_intkey_transaction(
+                verb="set",
+                name=name,
+                value=0,
+                deps=[],
+                private_key=self._private_key,
+                public_key=self._public_key)
+
+            batch = create_batch(
+                transactions=[txn],
+                private_key=self._private_key,
+                public_key=self._public_key)
+
+            self._deps[name] = txn.header_signature
+            batch_id = batch.header_signature
+
+            batch_list = batch_pb2.BatchList(batches=[batch])
+            stream.send(
+                message_type=Message.CLIENT_BATCH_SUBMIT_REQUEST,
+                content=batch_list.SerializeToString())
+
+            with self._lock:
+                self._pending_batches[batch_id] = \
+                    IntKeyState(name=name, stream=stream, value=0)
+
+            self.delegate.on_new_batch(batch_id, stream)
+
+
+def do_workload(args):
+    """
+    Create WorkloadGenerator and IntegerKeyWorkload. Set intkey workload in
+    generator and run.
+    """
+    generator = WorkloadGenerator(args)
+    workload = IntegerKeyWorkload(generator, args)
+    generator.set_workload(workload)
+    generator.run()
+
+
+def add_workload_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser(
+        'workload',
+        parents=[parent_parser],
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument('--rate',
+                        type=int,
+                        help='Batch rate in batches per minute. '
+                             'Should be greater then 0.',
+                        default=10)
+    parser.add_argument('-d', '--display-frequency',
+                        type=int,
+                        help='time in seconds between display of batches '
+                             'rate updates.',
+                        default=30)
+    parser.add_argument('-u', '--urls',
+                        help='comma separated urls of validators to connect '
+                        'to.',
+                        default="tcp://0.0.0.0:40000")
```

Comment:
I think the default should be "tcp://127.0.0.1:40000" or "tcp://127.0.0.1:40000". Typically 0.0.0.0 is used for listening on all addresses whereas the workload is sending packets at a specific address.
https://en.wikipedia.org/wiki/0.0.0.0
https://en.wikipedia.org/wiki/Localhost